File: 60.py
'''
@description: 查找组成一个偶数最接近的两个素数
@param {type} 
@return: 
'''
import math
 
def isPrime(n):
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            return False
    return True
 
 
while True:
    try:
        num ,start= int(input()) // 2,1
        if num%2==1:
            start=0
        for i in range(start, num, 2):
            a, b = num + i, num - i
            if isPrime(a) and isPrime(b):
                print(b)
                print(a)
                break
 
    except:
        break

File: 60.c

/**
 * @description: 查找组成一个偶数最接近的两个素数
 * @param {type} 
 * @return: 
 */
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <math.h>

int testnum;
int baocun1,baocun2,record;
int temp1, temp2,temp;
int issushu(int num)
{
    int temp = sqrt(num);
    for(int i = 2; i <= temp; i++)
    {
        if(num%i == 0)
        {
            return 0;
        }
    }
    return 1;
}

int main(void)
{
    while(scanf("%d", &testnum) != EOF)
    {
        temp = testnum;
        record = testnum;
        for(int i = 2; i <= temp; i++)
        {
            if(issushu(i)&&(issushu(testnum - i)))
            {
                if(abs(i - testnum + i) < record)
                {
                    record = abs(i - testnum + i);
                    baocun1 = i;
                    baocun2 = testnum - i;
                }
            }
        }
        printf("%d\n%d\n", baocun1, baocun2);
    }

    return 0;
}
